paraflow-metaserver/src/main/java/cn/edu/ruc/iir/paraflow/metaserver/utils/MetaConfig.java

```diff
@@ -58,7 +58,7 @@ public boolean validate()
             boolean user = mUser.find();
             //HDFSWarehouse format check
             String hdfsWarehouse = getHDFSWarehouse();
-            String regExWarehouse = "^[a-zA-Z]+://[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}:[0-9]{1,5}(/[a-zA-Z]+)+$";
+            String regExWarehouse = "^[a-zA-Z]+://(([0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3})|[a-zA-Z]+):[0-9]]{0,5}(/[a-zA-Z]+)+$";
```

Comment:
extra `]` in regex. Cannot match all cases, for example: hdfs://presto01:9000/metabase